### PR TITLE
samples(adt): add snippets for get and deserialize digital twin

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs [hunk 1]
```diff
@@ -32,7 +33,7 @@ public async Task RunSamplesAsync()
 
             string componentModelId = await GetUniqueModelIdAsync(SamplesConstants.TemporaryComponentModelPrefix, DigitalTwinsClient).ConfigureAwait(false);
             string modelId = await GetUniqueModelIdAsync(SamplesConstants.TemporaryModelPrefix, DigitalTwinsClient).ConfigureAwait(false);
-            string dtId1 = await GetUniqueTwinIdAsync(SamplesConstants.TemporaryTwinPrefix, DigitalTwinsClient).ConfigureAwait(false);
+            string basicDtId = await GetUniqueTwinIdAsync(SamplesConstants.TemporaryTwinPrefix, DigitalTwinsClient).ConfigureAwait(false);
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs:36]: @vinagesh I renamed these while looking at the doc comments. Out of context the dt1 and dt2 didn't make any sense. What do you think?

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/Readme.md [hunk 2]
```diff
@@ -52,9 +53,9 @@ var dtClient = new DigitalTwinsClient(
     clientOptions);
 ```
 
-## Create, List, Decommision and Delete Models
+## Create, list, decommision, and delete models
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/Readme.md:56]: decommission

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs (hunk 1 above)
Comment (vinagesh) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs:36]: I like this much better! Easier to understand the concept :)

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/Readme.md (hunk 2 above)
Comment (vinagesh) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/Readme.md:56]: Is this British/American English spelling thing or David really has a typo :P

**Review by vinagesh — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs [hunk 3]
```diff
@@ -66,19 +67,45 @@ public async Task RunSamplesAsync()
...
 
+            // You can also get a digital twin and deserialize it into a BasicDigitalTwin.
+            // It works well for basic stuff, but as you can see it gets more difficult when delving into
+            // more complex properties, like components.
+
+            #region Snippet:DigitalTwinsSampleGetBasicDigitalTwin
+
+            Response<string> getBasicDtResponse = await DigitalTwinsClient.GetDigitalTwinAsync(basicDtId).ConfigureAwait(false);
+            if (getBasicDtResponse.GetRawResponse().Status == (int)HttpStatusCode.OK)
+            {
```
Comment (prmathur-microsoft) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs:85]: What about else scenarios? Don't we want used to know it failed.

**Review by prmathur-microsoft — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs (hunk 3 above)
Comment (drwill-ms) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs:85]: If I had something compelling to put in the sample for that, I would. We have lots of other examples of errors in other places. I tried to sculpt this sample best for the doc comment with transform to web documentation in mind.

**Review by barustum — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/Readme.md (hunk 2 above)
Comment (drwill-ms) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/Readme.md]: It wasn't my typo

**Review by azabbasi — APPROVED** [review]

**Pull request merged**